stmhal/rtc.c

```diff
@@ -52,119 +54,119 @@ static mp_uint_t rtc_info;
 
 // Note: LSI is around (32KHz), these dividers should work either way
 // ck_spre(1Hz) = RTCCLK(LSE) /(uwAsynchPrediv + 1)*(uwSynchPrediv + 1)
+// modify RTC_ASYNCH_PREDIV & RTC_SYNCH_PREDIV in board/<BN>/mpconfigport.h to change sub-second ticks
+// default is 3906.25 µs, min is ~30.52 µs (will increas Ivbat by ~500nA)
+#ifndef RTC_ASYNCH_PREDIV
 #define RTC_ASYNCH_PREDIV (0x7f)
+#endif
+#ifndef RTC_SYNCH_PREDIV
 #define RTC_SYNCH_PREDIV  (0x00ff)
+#endif
 
-#if 0
-#define RTC_INFO_USE_EXISTING (0)
-#define RTC_INFO_USE_LSE (1)
-#define RTC_INFO_USE_LSI (3)
-
-void rtc_init(void) {
-    // Enable the PWR clock
-    RCC_APB1PeriphClockCmd(RCC_APB1Periph_PWR, ENABLE);
-
-    // Allow access to RTC
-    PWR_BackupAccessCmd(ENABLE);
-
-    if (RTC_ReadBackupRegister(RTC_BKP_DR0) == 0x32F2) {
-        // RTC still alive, so don't re-init it
-        // wait for RTC APB register synchronisation
-        RTC_WaitForSynchro();
-        rtc_info = RTC_INFO_USE_EXISTING;
-        return;
+/*
+ * lse_magic
+ *
+ */
+bool lse_magic(void) {
```

Comment:
Will this function work if load caps are connected to C14/C15 but no crystal?  Or does it require the pins to be floating?